src/main/java/core/basesyntax/SalaryInfo.java

```
@@ -1,7 +1,57 @@
+    private static final int WORKING_HOUR = 2;
+    private static final int INCOME_PER_HOUR = 3;
+    private static final String SEPARATOR = " - ";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period "
+                + dateFrom + SEPARATOR + dateTo + System.lineSeparator());
+        int count = 0;
+        for (String name : names) {
+            int earnedMoneyByPerson = getEarnedMoney(data, name, dateFrom, dateTo);
+            if (count == names.length - 1) {
+                result.append(name).append(SEPARATOR).append(earnedMoneyByPerson);
+            } else {
+                result.append(name).append(SEPARATOR).append(earnedMoneyByPerson)
+                        .append(System.lineSeparator());
+                count++;
+            }
+        }
+        return result.toString();
+    }
+
+    private int getEarnedMoney(String[] data, String nameOfPerson, String dateFrom, String dateTo) {
+        int sum = 0;
+        for (String elementOfData : data) {
+            String[] detailsOfElementOfData = elementOfData.split(" ");
+            if (detailsOfElementOfData[NAME_OF_PERSON].equals(nameOfPerson)) {
+                LocalDate particularData = null;
+                LocalDate fromDate = null;
+                LocalDate toDate = null;
+                try {
+                    particularData = LocalDate.parse(detailsOfElementOfData[DATA], FORMATTER);
+                    fromDate = LocalDate.parse(dateFrom, FORMATTER);
+                    toDate = LocalDate.parse(dateTo, FORMATTER);
+                    fromDate = fromDate.minusDays(1);
+                    toDate = toDate.plusDays(1);
+                } catch (DateTimeParseException e) {
+                    e.printStackTrace();
```

bad approach, it's better to throw an exception with informative message